dev/core/src/com/google/gwt/dev/DevMode.java

```diff
@@ -122,24 +125,48 @@ public boolean setFlag(boolean value) {
   }
 
   /**
-   * Handles the -server command line flag.
+   * Handles the -server command line flag. If unspecified, tries to find a single SCL defined in
+   * the service loader, or else defaults to StaticResourceServer.
    */
   protected static class ArgHandlerServer extends ArgHandlerString {
 
-    private static final String DEFAULT_SCL = JettyLauncher.class.getName();
-
-    private HostedModeOptions options;
+    private static final String DEFAULT_SCL = StaticResourceServer.class.getName();
 
+    private final HostedModeOptions options;
+    private final Map<String, ServletContainerLauncher> registered;
     public ArgHandlerServer(HostedModeOptions options) {
       this.options = options;
+      registered = ServiceLoader.load(ServletContainerLauncher.class).stream()
+          .map(ServiceLoader.Provider::get)
+          .filter(scl -> {
+            if (!ServletContainerLauncher.SERVICE_NAME_PATTERN.matcher(scl.getName()).matches()) {
+              System.err.println("Server class '" + scl.getClass().getName() +
+                  "' has an invalid name '" + scl.getName() +
+                  "'. To be used from the service loader, this name must match " +
+                  ServletContainerLauncher.SERVICE_NAME_PATTERN.pattern() + ". Skipping.");
+              return false;
+            }
+            return true;
+          })
+          .collect(Collectors.toMap(
+              ServletContainerLauncher::getName,
+              scl -> scl));
     }
 
     @Override
     public String[] getDefaultArgs() {
       if (options.isNoServer()) {
         return null;
       } else {
-        return new String[] {getTag(), DEFAULT_SCL};
+        if (registered.size() == 1) {
+          // Exactly one registered SCL, use it as the default, by fully qualified class name
+          return new String[] {
+              getTag(),
+              registered.values().iterator().next().getClass().getName()
+          };
```

Comment:
Would it make sense here to log that we either found none or found multiple launchers and that is why we revert to the default one.

Reply:
Maybe just return empty string here and make sure `""` checks for service loader first in `setString`?